patches/android_frameworks_base-R.patch

```diff
@@ -0,0 +1,177 @@
+
+Change-Id: Ied7d6ce0b83a2d2345c3abba0429998d86494a88
+---
+ api/current.txt                               |  2 ++
+ core/res/AndroidManifest.xml                  | 15 ++++++++++++
+ core/res/res/values/config.xml                |  2 ++
+ core/res/res/values/strings.xml               | 12 ++++++++++
+ non-updatable-api/current.txt                 |  2 ++
+ .../server/pm/PackageManagerService.java      | 23 +++++++++++++++++--
+ 6 files changed, 54 insertions(+), 2 deletions(-)
+
+diff --git a/api/current.txt b/api/current.txt
+index 4ab72254811..b3ad0711b95 100644
+--- a/api/current.txt
++++ b/api/current.txt
+@@ -79,6 +79,7 @@ package android {
+     field public static final String DUMP = "android.permission.DUMP";
+     field public static final String EXPAND_STATUS_BAR = "android.permission.EXPAND_STATUS_BAR";
+     field public static final String FACTORY_TEST = "android.permission.FACTORY_TEST";
++    field public static final String FAKE_PACKAGE_SIGNATURE = "android.permission.FAKE_PACKAGE_SIGNATURE";
+     field public static final String FOREGROUND_SERVICE = "android.permission.FOREGROUND_SERVICE";
+     field public static final String GET_ACCOUNTS = "android.permission.GET_ACCOUNTS";
+     field public static final String GET_ACCOUNTS_PRIVILEGED = "android.permission.GET_ACCOUNTS_PRIVILEGED";
+@@ -183,6 +184,7 @@ package android {
+     field public static final String CALL_LOG = "android.permission-group.CALL_LOG";
+     field public static final String CAMERA = "android.permission-group.CAMERA";
+     field public static final String CONTACTS = "android.permission-group.CONTACTS";
++    field public static final String FAKE_PACKAGE = "android.permission-group.FAKE_PACKAGE";
+     field public static final String LOCATION = "android.permission-group.LOCATION";
+     field public static final String MICROPHONE = "android.permission-group.MICROPHONE";
+     field public static final String NETWORK = "android.permission-group.NETWORK";
+diff --git a/core/res/AndroidManifest.xml b/core/res/AndroidManifest.xml
+index 2144d893a1f..91952adf6d4 100644
+--- a/core/res/AndroidManifest.xml
++++ b/core/res/AndroidManifest.xml
+@@ -2849,6 +2849,21 @@
+         android:description="@string/permdesc_getPackageSize"
+         android:protectionLevel="normal" />
+ 
++    <!-- Dummy user-facing group for faking package signature -->
++    <permission-group android:name="android.permission-group.FAKE_PACKAGE"
++        android:label="@string/permgrouplab_fake_package_signature"
++        android:description="@string/permgroupdesc_fake_package_signature"
++        android:request="@string/permgrouprequest_fake_package_signature"
++        android:priority="100" />
++
++    <!-- Allows an application to change the package signature as
++         seen by applications -->
++    <permission android:name="android.permission.FAKE_PACKAGE_SIGNATURE"
++        android:permissionGroup="android.permission-group.UNDEFINED"
```

Comment:
Line 70 needs to be 
`android:permissionGroup="android.permission-group.FAKE_PACKAGE"`
otherwise strings will not be correct in GUI.